### DISCUSSION TITLE
How should I use web3modal to integrate with web3.js??

### DISCUSSION THREAD

**Discussion by liubin595338764** [WalletConnect/walletconnect-monorepo#2972] [Developer Support]
https://docs.walletconnect.com/2.0/web/web3modal/html/wagmi/installation

I carefully reviewed this document and couldn't find a provider that can be used for web3.js calls.

If you use `@walletconnect/ethereum-provider`, you can do it like this:
```
const provider = await EthereumProvider.init({
      projectId: 'projectId'
      chains: [1], // REQUIRED chain ids
      showQrModal: true,
})

await provider.enable();
web3js = new Web3(provider);  
```
But if I use `@web3modal/html`, how can I generate a `provider` that can be used by `new Web3(provider)`? ？

**Comment by crypblizz8** [comment]
Hey there, here are all of our w3m examples: https://github.com/WalletConnect/web3modal-examples

Does that help you with next steps?

**Reply by liubin595338764** [reply]
@crypblizz8  Thank you, but this is not helpful. 
I have already gone through all the code here, and I am not using the React framework. 
What I need is a combination of HTML+web3modal+web3.js. Currently, the web3modal version in HTML does not have a valid provider for web3.js to use.

**Comment by glitch-txs** [comment]
Wagmi ethers adapter can help you, you can replace ethers with web3.js

**Reply by liubin595338764** [reply]
@glitch-txs hi, Wagmi is based on React, but our project is HTML. We need  HTML in order to use them.

**Reply by glitch-txs** [reply]
Wagmi has two packages, one based on react and another one in vanilla (core package)

Web3modal/html works with wagmi/core

**Reply by liubin595338764** [reply]
@glitch-txs  oh, Thank you, I found it.

I also found another problem:
may I ask how `provider` is passed to `wagmi`, for example:
```
const provider = await EthereumProvider.init({
    projectId,
    chains: [5], // REQUIRED chain ids
    showQrModal: true,
    relayUrl: "wss://relay.walletconnect.org",
})
```
After linking the wallet, I don't know how to pass it to wagmi.

**Reply by glitch-txs** [reply]
That's another package.

Check dependencies
https://docs.walletconnect.com/2.0/web/web3modal/html/wagmi/installation

**Reply by glitch-txs** [reply]
If you're using ethereum-provider package then you just pass the provider to web3.js but they are different packages

**Reply by liubin595338764** [reply]
@glitch-txs  Hello, if I want to be compatible with Ethereum/Tron and also require Sign functionality using web3.js, do I have to use `@walletconnect/sign-client`? How should I generate the provider required by web3.js? Is there a method that can handle signing for both Ethereum and Tron simultaneously?
